packages/pn-personagiuridica-webapp/src/redux/dashboard/actions.ts

```diff
@@ -4,27 +4,62 @@ import {
   formatToTimezoneString,
   getEndOfDay,
   getStartOfDay,
-  performThunkAction,
+  parseError,
 } from '@pagopa-pn/pn-commons';
 import { createAsyncThunk } from '@reduxjs/toolkit';
 
-import { NotificationsApi } from '../../api/notifications/Notifications.api';
+import { NotificationReceivedApiFactory, NotificationStatus } from '../../generated-client/notifications';
+import { apiClient } from '../../api/apiClients';
 
 export enum DASHBOARD_ACTIONS {
   GET_RECEIVED_NOTIFICATIONS = 'getReceivedNotifications',
 }
 
-export const getReceivedNotifications = createAsyncThunk<
-  GetNotificationsResponse,
-  GetNotificationsParams<Date> & { isDelegatedPage: boolean }
->(
+/**
+ * Get received notifications
+ */
+export const getReceivedNotifications = createAsyncThunk(
   DASHBOARD_ACTIONS.GET_RECEIVED_NOTIFICATIONS,
-  performThunkAction((params: GetNotificationsParams<Date> & { isDelegatedPage: boolean }) => {
-    const apiParams = {
-      ...params,
-      startDate: formatToTimezoneString(getStartOfDay(params.startDate)),
-      endDate: formatToTimezoneString(getEndOfDay(params.endDate)),
-    };
-    return NotificationsApi.getReceivedNotifications(apiParams);
-  })
+  async (params: GetNotificationsParams<Date> & { isDelegatedPage: boolean }, { rejectWithValue }) => {
+    try {
+      const receivedNotificationsFactory = NotificationReceivedApiFactory(
+        undefined,
+        undefined,
+        apiClient
+      );
+      const apiParams = {
+        ...params,
+        startDate: formatToTimezoneString(getStartOfDay(params.startDate)),
+        endDate: formatToTimezoneString(getEndOfDay(params.endDate)),
+        status: params.status as NotificationStatus | undefined
```

Comment:
Here we need the formatFiscalCode(params.recipientId) for recipient id filter. Copy from pf